util.c

```diff
@@ -324,10 +325,11 @@ static int keep_sockalive(SOCKETTYPE fd)
 
 	if (!opt_delaynet)
 #ifndef __linux
-		if (unlikely(setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, (const void *)&tcp_one, sizeof(tcp_one))))
+		unlikely_test = unlikely(setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, (const void *)&tcp_one, sizeof(tcp_one)));
 #else /* __linux */
-		if (unlikely(setsockopt(fd, SOL_TCP, TCP_NODELAY, (const void *)&tcp_one, sizeof(tcp_one))))
+		unlikely_test = unlikely(setsockopt(fd, SOL_TCP, TCP_NODELAY, (const void *)&tcp_one, sizeof(tcp_one)));
 #endif /* __linux */
+		if (unlikely_test)
```

Comment:
Also, this is broken. Notice the conditional on opt_delaynet before the #ifndef

You'll need to add braces when breaking up the one-liner.